tests/test_classical_mis.py

```diff
@@ -1,13 +1,71 @@
 from typing import Callable
 import networkx as nx
 import pytest
+from pathlib import Path
 
 # Define classical solver configuration
 from mis.solver.solver import MISInstance, MISSolver
 from mis.pipeline.config import SolverConfig
 from mis.shared.types import MethodType
 from mis.pipeline.kernelization import Kernelization
 
+TEST_FILES_DIR = Path.cwd() / "tests/test_files"
+
+
+@pytest.mark.parametrize("dimacs_to_nx", [TEST_FILES_DIR / "a265032_1tc.32.txt"], indirect=True)
+@pytest.mark.parametrize("preprocessor", [None, lambda graph: Kernelization(graph)])
+def test_for_dimacs_32_node_graph(
+    dimacs_to_nx: nx.Graph, preprocessor: None | Callable[[nx.Graph], Kernelization]
+) -> None:
+    """
+    Classical MIS solver for a standard graph benchmark dataset in DIMACS format.
+
+    Can be found here: https://oeis.org/A265032/a265032.html
+    """
+    graph = dimacs_to_nx
+    assert graph.number_of_nodes() == 32
+    assert graph.number_of_edges() == 68
+
+    config = SolverConfig(method=MethodType.EAGER, max_iterations=1, preprocessor=None)
+
+    # Create the MIS instance
+    instance = MISInstance(graph)
+
+    # Run the solver
+    solver = MISSolver(instance, config)
+    solutions = solver.solve().result()
+
+    assert len(solutions[0].nodes) == 12
+
+
+@pytest.mark.parametrize("dimacs_to_nx", [TEST_FILES_DIR / "a265032_1dc.64.txt"], indirect=True)
+@pytest.mark.parametrize("preprocessor", [None, lambda graph: Kernelization(graph)])
+def test_for_dimacs_64_node_graph(
+    dimacs_to_nx: nx.Graph, preprocessor: None | Callable[[nx.Graph], Kernelization]
+) -> None:
+    """
+    Classical MIS solver for a standard graph benchmark dataset in DIMACS format.
+
+    Can be found here: https://oeis.org/A265032/a265032.html
+    """
+    graph = dimacs_to_nx
+    assert graph.number_of_nodes() == 64
+    assert graph.number_of_edges() == 543
+
+    config = SolverConfig(method=MethodType.EAGER, max_iterations=1, preprocessor=preprocessor)
+
+    # Create the MIS instance
+    instance = MISInstance(graph)
+
+    # Run the solver
+    solver = MISSolver(instance, config)
+    solutions = solver.solve().result()
+
+    if preprocessor is None:
+        assert solutions[0].nodes == [1, 4, 42, 11, 16, 56, 25, 61]
+    else:
+        assert solutions[0].nodes == [64, 1, 4, 13, 16, 22, 47, 49, 52]
```

Comment:
Same problem here.

Also... we make no guarantee on the order of nodes in the solution, so perhaps compare sets?

Also... is it expected that the preprocessor affects the list of nodes?

Reply:
> Same problem here.
> 

OK will make that flaky.

> Also... we make no guarantee on the order of nodes in the solution, so perhaps compare sets?
> 

Fair, but I haven't got a solution from the dataset itself so I used what it seems to return _most of the time_, which feels kinda fragile.

> Also... is it expected that the preprocessor affects the list of nodes?

I don't know, you tell me.

Reply:
> Fair, but I haven't got a solution from the dataset itself so I used what it seems to return most of the time, which feels kinda fragile.

Yeah, that's the reason for which I'm only testing the size, but I'm not going to pretend that this is a great solution.

> I don't know, you tell me.

I suppose that it can, but it shouldn't :sweat:

Reply:
I think that we should find better criteria.